src/join.jl

```diff
@@ -343,18 +343,14 @@ function Base.join(f, left::Dataset, right::Dataset;
         lnulls[lnull_idx] .= true
         lout = if lout isa Columns
             Columns(map(lout.columns) do col
-                        if col isa DataValueArray
-                            col.isna[lnull_idx] .= true
-                        else
-                            DataValueArray(col, lnulls)
-                        end
-                    end)
+                v = Vector{Union{Missing,eltype(col)}}(col)
```

Comment:
Given that the old code modified the data in-place, I guess it's OK to avoid a copy? In that case using `convert` would be more appropriate.

Reply:
Judging by `@btime`, it looks like `convert` also creates a copy.  Should that not be the case?

Reply:
`convert` makes a copy if necessary, but avoids it if possible. So `col` is already of the target type, it will save a copy. (Also in the future it might reuse the values and only add a type tag to denote missingness, but currently that's not the case.)